hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java

```diff
@@ -114,6 +117,62 @@ public void whenStrategyClassNameNotExist_thenFailFast() {
     }
 
     @Test
+    public void givenDiscoveryStrategyFactoryExistOnClassPath_whenTheSameFactoryIsConfiguredExplicitly_thenOnlyOneInstanceOfStrategyIsCreated() {
+        Config config = new Config();
+        config.setProperty(GroupProperty.DISCOVERY_SPI_ENABLED.getName(), "true");
+
+        JoinConfig join = config.getNetworkConfig().getJoin();
+        join.getMulticastConfig().setEnabled(false);
+        DiscoveryConfig discoveryConfig = join.getDiscoveryConfig();
+
+        DiscoveryStrategyConfig strategyConfig = new DiscoveryStrategyConfig(new ParametrizedDiscoveryStrategyFactory());
+        strategyConfig.addProperty("bool-property", true);
+        discoveryConfig.addDiscoveryStrategyConfig(strategyConfig);
+
+        createHazelcastInstance(config);
+    }
```

Comment:
Are we missing the assertion on the `INSTANCE_COUNTER` here? I see no check of anything, beside no exception is thrown.

Reply:
see https://github.com/hazelcast/hazelcast/pull/11351/files/c8b284daf7bedbf6dc2dbcbaf15741f5add6ba1b#diff-2933953090c21972c398a529439c24f8R144

Reply:
the exception prevent Hazelcast instance to start -> failure

Reply:
Oh, that's well hidden :D

Reply:
good point anyway. I'll add a comment